specification/monitor/data-plane/Microsoft.Insights/stable/2023-10-01/metricBatch.json

```diff
@@ -0,0 +1,272 @@
+          },
+          {
+            "$ref": "#/parameters/MetricBatchResourceIdListParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The response to a metrics query.",
+            "schema": {
+              "$ref": "#/definitions/BatchMetricResultsResponse"
+            }
+          },
+          "default": {
+            "description": "Describes the format of Error response.",
+            "headers": {
+              "x-ms-error-code": {
+                "type": "string",
+                "description": "Error code for specific error that occurred."
+              }
+            },
+            "schema": {
+              "$ref": "../../../../resource-manager/Microsoft.Insights/stable/2023-10-01/commonMonitoringTypes.json#/definitions/ErrorContract"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Query Storage Metrics": {
+            "$ref": "./examples/MetricBatchExample.json"
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "MetricBatchSubscriptionIdParameter": {
+      "name": "subscriptionId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The subscription identifier for the resources in this batch.",
+      "x-ms-parameter-location": "method"
+    },
+    "MetricBatchResourceIdListParameter": {
+      "name": "resourceIds",
+      "in": "body",
+      "required": true,
+      "schema": {
+        "$ref": "#/definitions/MetricBatchResourceIdList"
+      },
+      "description": "The comma separated list of resource IDs to query metrics for."
+    },
+    "MetricBatchStartTimeParameter": {
+      "name": "starttime",
+      "in": "query",
+      "required": false,
+      "type": "string",
+      "description": "The start time of the query. It is a string in the format 'yyyy-MM-ddTHH:mm:ss.fffZ'. If you have specified the endtime parameter, then this parameter is required.\nIf only starttime is specified, then endtime defaults to the current time.\nIf no time interval is specified, the default is 1 hour.",
+      "x-ms-parameter-location": "method"
+    },
+    "MetricBatchEndTimeParameter": {
+      "name": "endtime",
+      "in": "query",
+      "required": false,
+      "type": "string",
+      "description": "The end time of the query. It is a string in the format 'yyyy-MM-ddTHH:mm:ss.fffZ'.",
+      "x-ms-parameter-location": "method"
+    },
+    "MetricBatchIntervalParameter": {
+      "name": "interval",
+      "in": "query",
+      "required": false,
+      "type": "string",
+      "default": "PT1M",
+      "description": "The interval (i.e. timegrain) of the query in ISO 8601 duration format. Defaults to PT1M. Special case for 'FULL' value that returns single datapoint for entire time span requested.\n*Examples: PT15M, PT1H, P1D*, FULL",
+      "x-ms-parameter-location": "method"
```

Comment:
Hey, @ToddKingMSFT, apologies if this was already discussed, but in previous versions, we also had `"format": "duration"` to describe interval parameters. Was there a reason this was removed? 

Asking since in the Python SDK generation, this was used to indicate that user-supplied `datetime.timedelta` objects should automatically be converted to the ISO 8601 duration format.

Reply:
Yeah that was removed because it fails model validation. There has been a long standing use of specifying "FULL" as a valid string to this interval parameter to indicate whatever the time range is the user just wants a single summary datapoint returned. This "FULL" string fails model validation when the format "duration" is used so we had to remove that was there is a LOT of use of this "FULL" string value for interval.

Reply:
Got it, thanks for the explanation!